creator-node/src/utils/legacyUtils.js

```diff
@@ -1,452 +0,0 @@
-const fs = require('fs-extra')
-const path = require('path')
-const axios = require('axios')
-const spawn = require('child_process').spawn
-const stream = require('stream')
-const { promisify } = require('util')
-
-const { logger: genericLogger } = require('../logging')
-const asyncRetry = require('./asyncRetry')
-const models = require('../models')
-const redis = require('../redis')
-const config = require('../config')
-const { generateTimestampAndSignature } = require('../apiSigning')
-const { libs } = require('@audius/sdk')
-
-const pipeline = promisify(stream.pipeline)
-const LibsUtils = libs.Utils
-
-const THIRTY_MINUTES_IN_SECONDS = 60 * 30
-
-export const EMPTY_FILE_CID = 'QmbFMke1KXqnYyBBWxB74N4c5SBnJMVAiMNRcGu6x1AwQH' // deterministic CID for a 0 byte, completely empty file
-
-export function verifySignature(data, sig) {
-  return recoverPersonalSignature({ data, sig })
-}
-
-export async function timeout(ms, log = true) {
-  if (log) {
-    genericLogger.info(`starting timeout of ${ms}`)
-  }
-  return new Promise((resolve) => setTimeout(resolve, ms))
-}
-
-/**
- * Generates a random number from [0, max)
- * @param {number} max the max random number. exclusive
- */
-export function getRandomInt(max) {
-  return Math.floor(Math.random() * max)
-}
-
-/**
- * Ensure DB and disk records exist for dirCID and its contents
- * Return fileUUID for dir DB record
- * This function does not do further validation since image_upload provides remaining guarantees
- */
-export async function validateStateForImageDirCIDAndReturnFileUUID(
-  req,
-  imageDirCID
-) {
-  // This handles case where a user/track metadata obj contains no image CID
-  if (!imageDirCID) {
-    return null
-  }
-  req.logger.info(
-    `Beginning validateStateForImageDirCIDAndReturnFileUUID for imageDirCID ${imageDirCID}`
-  )
-
-  // Ensure file exists for dirCID
-  const dirFile = await models.File.findOne({
-    where: {
-      multihash: imageDirCID,
-      cnodeUserUUID: req.session.cnodeUserUUID,
-      type: 'dir'
-    }
-  })
-  if (!dirFile) {
-    throw new Error(`No file stored in DB for imageDirCID ${imageDirCID}`)
-  }
-
-  // Ensure dir exists on disk
-  if (!(await fs.pathExists(dirFile.storagePath))) {
-    throw new Error(
-      `No dir found on disk for imageDirCID ${imageDirCID} at expected path ${dirFile.storagePath}`
-    )
-  }
-
-  const imageFiles = await models.File.findAll({
-    where: {
-      dirMultihash: imageDirCID,
-      cnodeUserUUID: req.session.cnodeUserUUID,
-      type: 'image'
-    }
-  })
-  if (!imageFiles) {
-    throw new Error(
-      `No image file records found in DB for imageDirCID ${imageDirCID}`
-    )
-  }
-
-  // Ensure every file exists on disk
-  await Promise.all(
-    imageFiles.map(async function (imageFile) {
-      if (!(await fs.pathExists(imageFile.storagePath))) {
-        throw new Error(
-          `No file found on disk for imageDirCID ${imageDirCID} image file at path ${imageFile.path}`
```

Comment:
I don't think `imageFile` has both a `storagePath` and a `path` so I changed it to just have `storagePath` in the refactored file

Reply:
yeah good find - `storagePath` is def correct